protocol.md

```diff
@@ -1078,13 +1078,24 @@ interface Hover {
 	contents: MarkedString | MarkedString[];
 	
 	/**
-	 * An optional range
+	 * An optional range is a range inside a text document 
+	 * that is used to visualize a hover, e.g. by changing the background color.
 	 */
 	range?: Range;
 }
 ```
 Where `MarkedString` is defined as follows:
 ```typescript
+/**
+ * The marked string is rendered:
+ * - as markdown if it is represented as a string
```

Comment:
Why forcing usage of markdown in the protocol when some other formats exists? Some servers may prefer HTML or asciidoc. Giving the choice to the server is actually a good thing, and if we want to give choice, then the MarkedString needs to remain to explicit the language; and if we have to keep MarkedString for that, why also keeping string?